Jenkinsfile.trigger

```diff
@@ -102,69 +102,98 @@ if (breakEarly) { return } // thanks Jenkins...
 
 // new data to be added to the past-jobs.json
 // { lastTime: unixTimestamp, url: "" }
-def buildCompletionData = [:]
+buildCompletionData = [:]
 
-for (buildObj in queue) {
-	stage(buildObj.identifier) {
-		//def json = writeJSON(json: buildObj, returnText: true)
-		//echo(json) // for debugging/data purposes
+// list of closures that we can use to wait for the jobs on.
+def waitQueue = [:]
+def addToWait(identifier, buildId, externalizableId) {
+	return {
+		// "catchError" to set "stageResult" :(
+		catchError(message: 'Build of "' + identifier + '" failed', buildResult: 'UNSTABLE', stageResult: 'FAILURE') {
+			stage(identifier) {
+				def res = waitForBuild(
+					runId: externalizableId,
+					propagateAbort: true, // allow cancelling this job to cancel all the triggered jobs
+				)
+				buildCompletionData[buildId] = [
+					lastTime: (res.startTimeInMillis + res.duration) / 1000, // convert to seconds
+					url: res.absoluteUrl,
+				]
+				if (res.result != 'SUCCESS') {
+					// set stage result via catchError
+					error(res.result)
+				}
+			}
+		}
+	}
+}
+
+// stage to wrap up all the build job triggers that get waited on later
+stage('trigger') {
+	for (int i = 0; i < queue.size(); i++) {
```

Comment:
Do you remember what the error is when you left this with the previous syntax so we could note it in a comment?

Reply:
I got it from reading https://www.jenkins.io/doc/pipeline/examples/#jobs-in-parallel and https://www.jenkins.io/doc/pipeline/examples/#parallel-from-grep, but it seems like the function is now enough to scope it correctly (tested locally), so I'll switch back to "for-in"